haystack/components/classifiers/zero_shot_document_classifier.py

```diff
@@ -0,0 +1,225 @@
+    pipeline.connect("retriever", "document_classifier")
+
+    queries = ["How was your day today?", "How was your day yesterday?"]
+    expected_predictions = ["positive", "negative"]
+
+    for idx, query in enumerate(queries):
+        result = pipeline.run({"retriever": {"query": query, "top_k": 1}})
+        assert result["document_classifier"]['documents'][0].to_dict()["id"] == str(idx)
+        assert (result["document_classifier"]['documents'][0].to_dict()["classification"]["label"]
+                == expected_predictions[idx])
+    ```
+    """
+
+    def __init__(
+        self,
+        model: str,
+        labels: List[str],
+        multi_label: bool = False,
+        classification_field: Optional[str] = None,
+        device: Optional[ComponentDevice] = None,
+        token: Optional[Secret] = Secret.from_env_var(["HF_API_TOKEN", "HF_TOKEN"], strict=False),
+        huggingface_pipeline_kwargs: Optional[Dict[str, Any]] = None,
+    ):
+        """
+        Initializes the TransformersZeroShotDocumentClassifier.
+
+        Available models for the task of zero-shot-classification include:
+        - ``'valhalla/distilbart-mnli-12-3'``
+        - ``'cross-encoder/nli-distilroberta-base'``
+
+        See https://huggingface.co/models for full list of available models.
+        Filter for zero-shot classification models (NLI): https://huggingface.co/models?pipeline_tag=zero-shot-classification&sort=downloads&search=nli
+
+        :param model:
+            The name or path of a Hugging Face model for zero shot document classification.
+        :param labels:
+            The set of possible class labels to classify each document into, e.g.,
+            ["positive", "negative"] otherwise None. Given a LABEL, the sequence fed to the model is "<cls> sequence to
+            classify <sep> This example is LABEL . <sep>" and the model predicts whether that sequence is a
+            contradiction or an entailment.
+        :param multi_label:
+            Whether or not multiple candidate labels can be true.
+            If `False`, the scores are normalized such that
+            the sum of the label likelihoods for each sequence is 1. If `True`, the labels are considered
+            independent and probabilities are normalized for each candidate by doing a softmax of the entailment
+            score vs. the contradiction score.
+        :param classification_field:
+            Name of document's meta field to be used for classification.
+            If left unset, `Document.content` is used by default.
+        :param device:
+            The device on which the model is loaded. If `None`, the default device is automatically
+            selected. If a device/device map is specified in `huggingface_pipeline_kwargs`, it overrides this parameter.
+        :param token:
+            The API token used to download private models from Hugging Face. If `token` is set to `True`, the token
```

Comment:
Docstring needs to be updated - the token can't be set to a boolean value (locally-stored secrets are not supported).